### PR TITLE
Issimagedrv

### PR DESCRIPTION
**Pull request by person124**
As I stated in the email it doesn't entirely work. Sorry about not doing the pull request until now I've been a mix of busy and tired.

### CODE REVIEW

**Comment by smokhov** [comment]
@rosserMartinez -- best commit message ever, but what does it do inside the commit?

File: src/api/c/quiz3/notes.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 Compiles in:
```
Comment (smokhov) [inline src/api/c/quiz3/notes.txt]: Need to move it to a section in `doc/CSI230F17.md` file with more details on tests, limitations and instructions for this specific driver.

File: src/api/c/quiz3/main.c [hunk 2]
```diff
@@ -15,6 +16,53 @@
...
 #define FILE_NAME_2 "/home/rosser/atsm/examples/c/syscalls/file/ifv/Makefile.orig"
 
+void hex_to_image(char outputf[], int hexSize, char hexdump[])
+{
+	int fd;
+	char num [13];
+	int iter = 0;
+
+	/*Create the file with the given ouput filename*/
+	fd = open(outputf, O_RDWR | O_CREAT, S_IRUSR | S_IRGRP | S_IROTH);
```
Comment (smokhov) [inline src/api/c/quiz3/main.c]: `fd` is never checked for the return value from the system call. Particularly bad idea inside a driver if it fails...

File: src/api/c/quiz3/main.c [hunk 3]
```diff
@@ -15,6 +16,53 @@
...
+	width = hexSize / 2;
+	
+	if(hexSize % 2 == 0)
+		width = hexSize / 2;
+	else
+		width = (hexSize / 2) -1;	
+	
+	/* Input the header and image size to the file */
+	snprintf(num, 10, "P3\n%i %i\n", width, height);
+	write(fd, num, strlen(num));
```
Comment (smokhov) [inline src/api/c/quiz3/main.c]: No return value check from `write` either...

**Comment by rosserMartinez** [comment]
@smokhov IT should be something insightful towards the change in function, like "now forks and reads / writes from multiple files at once now" but I totally kept trying to push without committing anything

Realized I hadn't committed and after the disbelief that ensued, I think that was the only message I could muster the energy to type

File: src/api/c/quiz3/main.c [hunk 4]
```diff
@@ -6,106 +6,122 @@
...
 #include <stdlib.h>
 #include <string.h>
+#include <math.h>
 
 #ifndef HEXDUMP_COLS
 #define HEXDUMP_COLS 8
 #endif
 
 #define DEVICE_NAME "/dev/ISSIMAGEDRV"
+#define FILE_NAME_1 "/boot/initramfs-4.14.0-1.el6.elrepo.x86_64.img"
```
Comment (smokhov) [inline src/api/c/quiz3/main.c]: Better try something that is more guaranteed to be present on the file system, like own `Makefile` or `main.c`...

**Pull request merged**